libs/ui-components/src/multi-select/MultiSelect.tsx

```diff
@@ -279,18 +281,21 @@ export function MultiSelect(props: MultiSelectProps) {
             filteredItems.map((item, index) => {
               const highlight = highlightedIndex === index
               const isSelected =
-                hasSelectedOptions && selectedOptions.some(({ id }) => item.id === id)
+                hasSelectedOptions &&
+                selectedOptions.some(({ id }) => item.id === id) &&
+                !item.disableSelection
               const icon =
                 highlight && isSelected
                   ? 'close'
-                  : highlight || isSelected
+                  : (highlight || isSelected) && !item.disableSelection
                   ? 'tick'
                   : ('' as IconType)
               return (
                 <Tooltip key={item.id} content={item.tooltip} placement="top-start">
                   <li
                     className={cx(styles.optionItem, {
                       [styles.highlight]: highlight,
+                      [item.className || '']: item.className,
```

Comment:
Don't understand this because the item.className will be only added when it exists
the fallback is useful when is used as:
`cx(styles[item.className])`
so this can be simplified as:
`[item.className]: item.className`

Reply:
Because `className` is optional in the component props, the TS compiler will complain that `A computed property name must be of type 'string', 'number', 'symbol', or 'any'.` This is IMHO the least bad solution, others include `[item.className as any]: item.className` or `[!item.className]: item.className`